H2MLauncher.UI/MainWindow.xaml.cs

```diff
@@ -110,5 +157,85 @@ private void DataGridRow_GotFocus(object sender, RoutedEventArgs e)
         {
             ((DataGridRow)sender).IsSelected = true;
         }
+
+        private void ShowOverlay()
+        {
+            _overlayHiddenByUser = false; // Mark overlay as shown by user
+            _isOverlayVisible = true; // Ensure overlay is visible
+            this.Visibility = Visibility.Visible;
+
+            // Optionally, you can update its position
+            UpdateOverlayPosition(null, null);
+        }
+        private void HideOverlay()
+        {
+            _overlayHiddenByUser = true; // Mark overlay as hidden by user
+            this.Visibility = Visibility.Hidden; // Simply hide the overlay
+        }
+
+        private void UpdateOverlayPosition(object sender, EventArgs e)
+        {
+            if (_targetWindowHandle == IntPtr.Zero) return;
+
+            if (IsIconic(_targetWindowHandle)) { this.Visibility = Visibility.Hidden; return; }
+
+            // Only update if not hidden by the user
+            if (!_overlayHiddenByUser)
+            {
+                // Show or hide the overlay based on the user's toggle action
+                this.Visibility = _isOverlayVisible ? Visibility.Visible : Visibility.Hidden;
+
+                // Set the overlay position with a default offset (you can adjust the offset value here)
+                SetOverlayPosition();
+            }
+        }
+
+        // Method to set the overlay position centered within the game window
+        private void SetOverlayPosition()
+        {
+            if (_targetWindowHandle == IntPtr.Zero || !GetWindowRect(_targetWindowHandle, out RECT rect))
+                return;
+
+            // Calculate the width and height of the game window
+            int gameWindowWidth = rect.Right - rect.Left;
+            int gameWindowHeight = rect.Bottom - rect.Top;
+
+            // Calculate position to center the overlay within the game window
+            int overlayWidth = (int)this.ActualWidth;
+            int overlayHeight = (int)this.ActualHeight;
+
+            int xPosition = rect.Left + (gameWindowWidth - overlayWidth) / 2;
+            int yPosition = rect.Top + (gameWindowHeight - overlayHeight) / 2;
+
+            // Move the WPF window to the calculated position
+            this.Left = xPosition;
+            this.Top = yPosition;
+        }
+
+        private void AttachToH2MModWindow()
+        {
+            _targetWindowHandle = FindWindow(null, "H2M-Mod");
+
+            if (_targetWindowHandle == IntPtr.Zero)
+            {
+                MessageBox.Show("H2M-Mod window not found.");
+                return;
+            }
+
+        }
+
+        private void GlobalKeyboardHook_KeyPressed(Key key, ModifierKeys modifiers)
+        {
+            if (key == Key.S && modifiers.HasFlag(ModifierKeys.Control) && modifiers.HasFlag(ModifierKeys.Alt))
```

Comment:
I think it might be wise to just keep the same combination of keys to flip its visibility. Just CTRL + ALT + S